### PR TITLE
extensions: use buttons for non-link actions

### LINKED ISSUE #34054: [Accessibility Audit] Integrations: Interacting with an extension on the search page
### Steps to replicate journey

Taken from the [user journey list document](https://docs.google.com/document/d/1_IGekkIDuQugFA-2BYYWpYWpJy9O8PjiDrEEGmJxRRU/edit?usp=sharing). Use this for further context.

- Enable the [search-export](https://sourcegraph.com/extensions/sourcegraph/search-export) extension.
- Go to the homepage, execute a search.
- Observe the "Sourcegraph: Export search results" button in the search results info bar.
- Check that clicking the button triggers an action.
  - Note: We cannot guarantee WCAG 2.1 compliance for the actions of these extensions, as they can be created by third-parties.

### How to audit

Follow the instructions here: [Auditing a user journey](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey)

**Note:** We are planning on arranging thorough third-party audit in the future, so our focus here is just to find the _majority_ of accessibility issues. Don't worry if you feel you haven't found 100% of potential issues, it is better to focus on the core essentials to complete the journey rather than spending lots of time going through every possible problem.

### How to raise discovered problems

Follow the instructions here: [Raising an accessibility bug](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#raising-a-bug)

### Any other questions?

See the [audit FAQ](https://docs.google.com/document/d/1nCi7TvYRpj9eG0_QtgqK1c4L28EiJKVuxyuvg1lAGgo/edit#) - add your question if it isn't there already!

### PR DESCRIPTION
**Pull request by tjkandala**
Part of #36751.
Closes #34054.

Looked like a codemod for `class="btn"` [wrapped `<ActionItem>`s in `<SearchResultsInfoBar>` with `<Button>`](https://github.com/sourcegraph/sourcegraph/pull/36917/files#diff-43c7ea1c1ea1857c1e22d85405cdf3581bf84c3461c48249874f7c638e134091L357). We used to pass that style down to `<ButtonLink>`, which renders `<a>`s for everything :(.   

I refactored `<ActionItem>` to render `<Button>` when the extension action isn't navigation. 

I also did some migration from `data-tooltip` to our new wildcard `<Tooltip>`. I found that wrapping action items with `<Tooltip>` affects styles due to insertion of the target `<span>`. This broke the adjacent sibling combinator (`+`) in hover overlay for buttons like "Find references" and "Find implementations". What is FPTs suggestion for such styles? I just added the `ml-1` class to hover overlay buttons that aren't the first in `<HoverOverlay>`, but I wonder if there's a pure CSS solution.

## Test plan

Manually tested all components that use `<ActionItem>` in both the webapp and browser extension, automated tests pass.

### Manual testing results

| Component | Before | After |
| ----------- | ----------- | -------- |
|  `<SearchResultsInfoBar>` | before-srib | after-srib |
| `<HoverOverlay>` | ho-before | ho-after |
| `<CommandList>` |  cl-before |  cl-after |
| `<ActionItemsBar>` | aib-before | aib-after |
| Action items on BExt | aibext-before | aibext-after |

### Changes

- Slight increase in command list font weight, but that aligns with our other buttons so it should be ok.
- Browser extension action items look better somehow? Might be the icon centering.

## App preview:

- [Web](https://sg-web-tjk-action-items-buttons.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-jfbryvvkhp.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/shared/src/actions/ActionItem.tsx [hunk 1]
```diff
@@ -250,56 +263,97 @@ export class ActionItem extends React.PureComponent<ActionItemProps, State, type
...
+        }
+
+        const tooltipOrErrorMessage =
+            this.props.showInlineError && isErrorLike(this.state.actionOrError)
+                ? `Error: ${this.state.actionOrError.message}`
+                : tooltip
+
+        if (!to) {
+            return (
+                <Tooltip className={this.props.tooltipClassName} content={tooltipOrErrorMessage ?? null}>
```
Comment (tjkandala) [inline client/shared/src/actions/ActionItem.tsx:291]:
```suggestion
                <Tooltip className={this.props.tooltipClassName} content={tooltipOrErrorMessage || null}>
```

File: client/shared/src/actions/ActionItem.tsx [hunk 2]
```diff
@@ -250,56 +263,97 @@ export class ActionItem extends React.PureComponent<ActionItemProps, State, type
...
-                {!this.props.hideExternalLinkIcon && primaryTo && isExternalLink(primaryTo) && (
-                    <OpenInNewIcon className={this.props.iconClassName} />
-                )}
-                {showLoadingSpinner && (
-                    <div className={styles.loader} data-testid="action-item-spinner">
-                        <LoadingSpinner inline={false} className={this.props.iconClassName} />
-                    </div>
-                )}
-            </ButtonLink>
+            <Tooltip className={this.props.tooltipClassName} content={tooltipOrErrorMessage ?? null}>
```
Comment (tjkandala) [inline client/shared/src/actions/ActionItem.tsx:319]:
```suggestion
            <Tooltip className={this.props.tooltipClassName} content={tooltipOrErrorMessage || null}>
```

File: client/shared/src/actions/ActionItem.tsx [hunk 3]
```diff
@@ -250,56 +263,97 @@ export class ActionItem extends React.PureComponent<ActionItemProps, State, type
...
+                <ButtonLink
+                    data-content={this.props.dataContent}
+                    disabled={
```
Comment (tjkandala) [inline client/shared/src/actions/ActionItem.tsx:322]: TODO: Use shared props

**Review by taras-yemets — APPROVED** [review]

**Comment by tjkandala** [comment]
Screen Shot 2022-06-21 at 4 30 47 PM

Bug: go to definition action didn't work on the browser extension on this branch. Reverting from the new `<Tooltip>` component to `data-tooltip` attribute fixed the issue.

**Pull request merged**